AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/AWSDataStorePlugin.swift

```diff
@@ -82,7 +88,7 @@ final public class AWSDataStorePlugin: DataStoreCategoryPlugin {
         let filter = HubFilters.forEventName(HubPayload.EventName.Amplify.configured)
         var token: UnsubscribeToken?
         token = Amplify.Hub.listen(to: .dataStore, isIncluded: filter) { _ in
-            if self.hasValidAPIPlugin() {
+            if self.hasRequiredPlugins() {
                 self.storageEngine.startSync()
             } else {
                 self.log.info("Unable to find suitable plugin for syncEngine.  syncEngine will not be started")
```

Comment:
I think we could be more informative here. Since the sync engine could not start for different reasons it would be cool if we provided a more detailed message of the actual reason it didn't start (and maybe use `log.warn`?)

Reply:
i consolidated this logic and the logic from RemoteSyncEngine over to StorageEngine and added more logging. I have some `warn` and some `error` as well now. the initial check (in the new code) checks for APIPlugin and that is an `info` because this is to account for developers using local store first without API. all other logging are `warn` afterwards where there is API but something else is unexpected